app/code/Magento/Catalog/Model/ResourceModel/Product/Indexer/Price/Query/BaseFinalPrice.php

```diff
@@ -223,8 +223,11 @@ public function getQuery(array $dimensions, string $productType, array $entityId
         $select->where("e.type_id = ?", $productType);
 
         if ($entityIds !== null) {
```

Comment:
This could be redundant because of strict typing and type=array of $entityIds

Reply:
@AndreyChorniy what do you think about making this improvement as well?

Reply:
Hello @hws47a, 
I have checked few usages for this method in Magento extensions, I see that param `$entityIds` is passed wrapped in function `iterator_to_array()`
For example [here](https://github.com/magento/magento2/blob/d69c242d9033616cd88018eea97f3a19bbc2da70/app/code/Magento/ConfigurableProduct/Model/ResourceModel/Product/Indexer/Price/Configurable.php#L135)
So as for me, it should not be changed. 

@hws47a What do you think about it?